### PR TITLE
Fixed two issues with GT annotations

### PR DESCRIPTION
**Pull request by bsekachev**
### Motivation and context
1. When there are conflicts on the first frame, they are not highlighted properly if CHANGE_FRAME_SUCCESS is not fired yet
2. UI crash if to hide some annotations related to a conflict and hover mouse over the conflict

issue 1
problem 2

### Checklist

- [x] I submit my changes into the `develop` branch
- [x] I have created a changelog fragment 
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] I have linked related issues (see [GitHub docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [x] I have increased versions of npm packages if it is necessary
  ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
  [cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning),
  [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and
  [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.

### CODE REVIEW

File: changelog.d/20231228_131558_boris_fixed_couple_of_gt_bugs.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+### Fixed
+
+- UI crashes if user highligts conflict related to annotations hidden by a filter
+  (<https://github.com/opencv/cvat/pull/XXXX>)
+- Annotations conflicts are not highligted properly on the first frame of a job
+  (<https://github.com/opencv/cvat/pull/XXXX>)
+
```
Comment (bsekachev) [inline changelog.d/20231228_131558_boris_fixed_couple_of_gt_bugs.md:7]:
```suggestion
  (<https://github.com/opencv/cvat/pull/7299>)
- Annotations conflicts are not highligted properly on the first frame of a job
  (<https://github.com/opencv/cvat/pull/7299>)
```

File: cvat-ui/src/components/annotation-page/review/issues-aggregator.tsx [hunk 2]
```diff
@@ -47,7 +47,7 @@ export default function IssueAggregatorComponent(): JSX.Element | null {
     const showConflicts = useSelector((state: CombinedState) => state.settings.shapes.showGroundTruth);
     const highlightedConflict = useSelector((state: CombinedState) => state.annotation.annotations.highlightedConflict);
 
-    const highlightedObjectsIDs = highlightedConflict?.annotationConflicts?.map((c: AnnotationConflict) => c.clientID);
+    const highlightedObjectsIDs = highlightedConflict?.annotationConflicts?.map((c: AnnotationConflict) => c.serverID);
```
Comment (klakhov) [inline cvat-ui/src/components/annotation-page/review/issues-aggregator.tsx:50]: Let's rename `c` to something more meaningful

File: cvat-ui/src/components/annotation-page/review/issues-aggregator.tsx [hunk 3]
```diff
@@ -127,30 +127,37 @@ export default function IssueAggregatorComponent(): JSX.Element | null {
...
-                            y: points[1],
-                            clientID: c.clientID,
-                            conflict,
-                        };
+        if (canvasReady && showConflicts && qualityConflicts.length) {
+            const updatedConflictMapping = qualityConflicts
+                .map((conflict: QualityConflict) => {
+                    const mainAnnotationsConflict = conflict.annotationConflicts[0];
+                    const state = objectStates.find((s: ObjectState) => (
+                        s.serverID === mainAnnotationsConflict.serverID &&
```
Comment (klakhov) [inline cvat-ui/src/components/annotation-page/review/issues-aggregator.tsx:135]: Same for S here

File: cvat-ui/src/components/annotation-page/canvas/views/canvas2d/canvas-wrapper.tsx [hunk 4]
```diff
@@ -481,12 +481,15 @@ class CanvasWrapperComponent extends React.PureComponent<Props> {
...
-                highlightedConflict?.severity ? (highlightedConflict?.severity as any) : null;
-            const highlightedElementsIDs = highlightedConflict?.annotationConflicts.map(
-                (conflict: AnnotationConflict) => conflict.clientID,
-            );
-            canvasInstance.highlight(highlightedElementsIDs || null, severity);
+            const severity: HighlightSeverity | undefined = highlightedConflict
+                ?.severity as unknown as HighlightSeverity;
+            const highlightedClientIDs = (highlightedConflict?.annotationConflicts || [])
+                .map((conflict: AnnotationConflict) => annotations
+                    .find((state) => state.serverID === conflict.serverID && state.objectType === conflict.type),
```
Comment (klakhov) [inline cvat-ui/src/components/annotation-page/canvas/views/canvas2d/canvas-wrapper.tsx:488]: `conflict.type` has a string type,  looks we need to change it to `ObjectType`

Comment (bsekachev) [reply cvat-ui/src/components/annotation-page/canvas/views/canvas2d/canvas-wrapper.tsx:488]: Done

**Review by klakhov — APPROVED** [review]

**Pull request merged**